cfme/common/provider_views.py

```diff
@@ -530,6 +542,19 @@ def is_displayed(self):
                 self.title.text == 'Edit Infrastructure Provider')
 
 
+class PhysicalProviderEditView(ProviderEditView):
+    """
+     represents Provider Edit View
+    """
+    @property
+    def is_displayed(self):
+        return (super(PhysicalProviderEditView, self).is_displayed and
+                self.navigation.currently_selected ==
+                ['Compute''Physical Infrastructure', 'Providers'] and
+                self.title.text == 'Edit Infrastructure Provider' and
+                self.context['object'].name == 'lenovo')
```

Comment:
So this isn't quite what I was looking for - this is just a check against the contextual object attribute itself, and isn't a check against anything in the UI.

Is there something in the UI, a widget of some type, that includes specific information about the provider being modified?

If not, this method should just return False with no other checks. This is a completely valid configuration, with the only downside being increased navigation times in cases where we could have avoided navigating at all.

If in doubt, better to return False until you've got a definite widget to validate against.